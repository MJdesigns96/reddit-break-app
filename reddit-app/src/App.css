.container {
    margin: 0 24px;
    font-family: Arial, Helvetica, sans-serif;
    background-color: rgb(255, 255, 255);
    padding: 0 48px;
}
