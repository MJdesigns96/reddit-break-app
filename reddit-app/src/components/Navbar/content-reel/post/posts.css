
.card {
    margin: 16px auto;
    width: 90%;
    border: .5;
}

div #upvote-counter {
    margin-left: 16px;
}

.card-title {
    text-decoration: none;
    color:black;
    margin-left: 16px;
    margin-top: 8px;
    margin-bottom: 0;
}

.card-title:hover {
    text-decoration: underline;
    color: black;
}

#preview {
    text-decoration: none;
    color: black;
    font-size: 32px;
    margin-bottom: 16px;
}

.card-body a {
    text-decoration: none;
    color: rgb(83, 83, 83);
}

.card-body a:hover {
    text-decoration: underline;
    color: black;
}