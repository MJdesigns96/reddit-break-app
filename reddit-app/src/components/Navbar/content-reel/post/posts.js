import React, { useContext, useEffect } from "react";
import moment from 'moment';

export const Post = (props) => {

    var utcSeconds = props.posts.created_utc;
    var time = moment.unix(utcSeconds).format("YYYY-MM-DD HH:mm");
    var timeFromNow = moment(time).fromNow();


    const preview = () => {
        if(props.posts.preview) {
            return props.posts.url;
        } else {
            return "https://www.iconpacks.net/icons/2/free-reddit-logo-icon-2436-thumb.png";
        }   
    }

    return (
        <div className="card" >
            <a href={ "https://reddit.com/" + props.posts.permalink }>
               <img className="card-img-top" src={preview()} width="1080"/> 
            </a>
            <div>
                <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-caret-up-fill" viewBox="0 0 16 16">
                    <path d="m7.247 4.86-4.796 5.481c-.566.647-.106 1.659.753 1.659h9.592a1 1 0 0 0 .753-1.659l-4.796-5.48a1 1 0 0 0-1.506 0z"/>
                </svg>
                { props.posts.score } Upvotes
            </div>
            <a className="card-title" href={ "https://reddit.com/" + props.posts.permalink } target="_blank">
                <h3>{ props.posts.title }</h3> 
            </a>
            <div className="card-body">
                <p>{ "Submitted by: " + props.posts.author } <br />{ "Posted: " +  timeFromNow}</p> 
            </div>
            
        </div>
    )
}

export default Post;
