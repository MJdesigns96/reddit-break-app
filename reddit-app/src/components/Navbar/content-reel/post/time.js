import React from "react";
import moment from 'moment';

export const Time = () => {
    const getTime = (props) => {
       var utcSeconds = props.posts.data.created_utc;
        var time = moment.unix(utcSeconds).format("YYYY-MM-DD HH:mm");
        var timeFromNow = moment(time).fromNow();

        return timeFromNow
    }

    

    return (
        <Time getTime={getTime} />
    )
    
}
