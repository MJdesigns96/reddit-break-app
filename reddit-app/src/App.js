import React, { useState, useEffect } from 'react';
import { Container } from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css';
import { Post } from './components/Navbar/content-reel/post/posts';

const app = () => {
  const [posts, setPosts] = useState([]);
  const [subreddit, setSubreddit] = useState('webdev');

  useEffect(()=> {
    fetch("https://www.reddit.com/r/baseball.json").then(response => {
      if (response.status != 200) {
        console.log("Error response from server unsuccessful");
        return;
      }

      response.json().then(data => {
        if (data != null) {
          setPosts(data.data.children);
        }
      });
    })
  }, [subreddit]);

  return (
    <Container>
      <div className='navbar'>
          <a className='navbar-brand' href={''}>
            <img src='https://www.iconpacks.net/icons/2/free-reddit-logo-icon-2436-thumb.png' width="32px" style={({marginRight: 8})} alt='logo' />
            Reddit Minimal
          </a> 
      </div>
      <div className='card'>
         {
          (posts != null) ? posts.map((post, index) => <Post key={index} posts={post.data} />) : ''
         }
      </div>
    </Container>
  )
}

export default app;
