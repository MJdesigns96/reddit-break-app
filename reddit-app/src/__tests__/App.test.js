import { render, screen } from '@testing-library/react';
import React from 'react';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from '../App';
import { Post } from '../components/Navbar/content-reel/post/posts';

configure({ adapter: new Adapter() });

it("renders a title", () => {
  const wrapper = shallow(
    <App />
  );
  expect(wrapper).toMatchSnapshot();
});

describe ('<Post />', () => {
  it ('renders a h3', () => {
    const wrapper = shallow(<Post />)
    console.log(wrapper.debug())
    expect(wrapper).toMatchSnapshot()
  })
})