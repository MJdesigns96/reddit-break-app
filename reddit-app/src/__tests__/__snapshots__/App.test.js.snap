// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a title 1`] = `ShallowWrapper {}`;
